modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1281,6 +1281,101 @@ CV__DNN_INLINE_NS_BEGIN
                              float confThreshold = 0.5f, float nmsThreshold = 0.0f);
      };
 
+     /** @brief This class represents high-level API for text recognition networks.
+      *
+      * SceneTextRecModel allows to set params for preprocessing input image.
+      * SceneTextRecModel creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return recognition result.
+      * For TextRecognitionModel, CRNN-CTC is supported.
+      */
+     class CV_EXPORTS_W TextRecognitionModel : public Model
+     {
+     public:
+         /**
+          * @brief Create text recognition model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+         CV_WRAP TextRecognitionModel(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from deep learning network.
+          * @param[in] network Net object.
+          */
+         CV_WRAP TextRecognitionModel(const Net& network);
+
+         /**
+          * @brief Set the vocabulary for recognition.
+          * @param[in] inputVoc: the associated vocabulary of the network.
+          */
+         CV_WRAP void setVocabulary(const std::vector<String>& inputVoc);
+
+         /**
+          * @brief Given the @p input frame, create input blob, run net and return recognition result.
+          * @param[in] frame: The input image.
+          * @param[in] decodeType: The decoding method of translating the network output into string: {'CTC-greedy': greedy decoding for the output of CTC-based methods}
+          * @param[out] results: A set of text recognition results.
+          * @param[in] roiPolygons: (Optional) A set of text detection results. If it is not empty, all the ROIs will be cropped as the network inputs.
+          */
+         CV_WRAP void recognize(InputArray frame, const String& decodeType, CV_OUT std::vector<String>& results,
```

Comment:
> decodeType

This is algorithm / model parameter.
(it is not intended to be changed between multiple "recognize" calls for different frames).

Must be moved into constructor / `::create()` methods and `.get` (may be `.set`) properties.

---

/cc @dkurt @l-bat @vpisarev

Reply:
I moved `decodeType` into `Voc`, but I have noticed that you propose to update cv::dnn::Model class in https://github.com/opencv/opencv/pull/18711.
Hence, I need to wait for that pr to be merged, and update my codes?